### DISCUSSION TITLE
F0036 - Keyhole optimiser is almost done

### DISCUSSION THREAD

**Discussion by adesutherland** [adesutherland/CREXX#164] [Progress Updates]
That's the good news.

The bad is I got the covid and our trip is cancelled! :-( Anyway it means I can make a session on Thursday

So the keyhole optimiser is now called the "covid-optimiser" or copt

A

**Comment by Peter-Jacob** [comment]
Best wishes for a speedy recovery!

**Comment by rvjansen** [comment]
Get well soon!

René.

**Comment by mgrossmann** [comment]
Get well, from me, too.

I‘m back in business….
Next session I will join.

See ya…

Mike

René Vincent Jansen ***@***.***> schrieb am So. 24. Okt. 2021
um 09:26:

> Get well soon!
>
> René.
>
> > On 23 Oct 2021, at 19:36, Adrian Sutherland ***@***.***> wrote:
> >
> > ﻿
> > That's the good news.
> >
> > The bad is I got the covid and our trip is cancelled! :-( Anyway it
> means I can make a session on Thursday
> >
> > So the keyhole optimiser is now called the "covid-optimiser" or copt
> >
> > A
> >
> > —
> > You are receiving this because you are subscribed to this thread.
> > Reply to this email directly, view it on GitHub, or unsubscribe.
> > Triage notifications on the go with GitHub Mobile for iOS or Android.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/adesutherland/CREXX/discussions/164#discussioncomment-1526870>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAF7Q7I5AUYZHGT6MIPB2FTUIOYI7ANCNFSM5GSPO7PA>
> .
>
-- 
Von Gmail Mobile gesendet

**Comment by adesutherland** [comment]
F0036 merged into develop 

See "documentation" here https://github.com/adesutherland/CREXX/wiki/RXAS-CovidOptimiser

It might introduce some issues ... rxas -n disables the covidoptimiser

Should I merge into f0035? 

Also, I notice that the CMS build in f0035 is failing, I hope we aren't reaching some memory limit on that platform :-(

**Comment by rvjansen** [comment]
Yes, please go ahead and merge in f0035. For CMS we might ask Bob to see whats wrong and tell us what to do.


René.

**Comment by BobBolch** [comment]
Hi everyone,
I will get home late tonight, so I can look into what is wrong in the
morning. I need someone to let me know where to look.

Bob

**Reply by adesutherland** [reply]
Rene ... I will merge in and the look at the CMS issues

Hello Bob! ... there seem to be a few minor issue that I will fix up. The main problem is that we may be running out of virtual memory for the main huge VM compile ...

I am going to play with GCC optimiser options, and restructuring :-(

Another option is to build it in a VM with as much memory available as possible. Perhaps by getting rid of BREXX / GCCLIB. Is there a user ID we can setup / or which exists we could use?

Anyway I am not quite ready to be definite about the problem yet!

Adrian

**Reply by rvjansen** [reply]
I built on macOS and note that I needed to scratch CMakeCache.txt in both my build directories - first it could not find <wchar.h>; fortunately this was a Stack Overflow documented situation.

**Reply by adesutherland** [reply]
I still can't find the reason for this :-(

I fear that reloading CMAKE (or deleting the cache) whenever the CMAKE files are changed will be necessary until we work it out ...

**Comment by BobBolch** [comment]
Hi Adrian,

It should be easy to have one of the CMSUSER IDs on your github VM system
to skip loading brexx. That might work.

Also, the port of the GCC compiler to CMS involved compiling steps that
took more than 16M. They used the VM/380 setup to get access to a large
memory so they could compile the compiler. I never looked into the specific
details, but that method might be used to compile crexx.

Bob

Also, I hope you get well soon!

**Reply by adesutherland** [reply]
I've just finished reading The Soul of a New Machine ... so I guess the s/380 concept is a 32 (31 I know) bit machine but with a mode bit!

Is it really possible to get a VM/380 working?

I am confident that that crexx VM and executables would fit into 24 bits, it's just the C compiler itself ... sigh!

**Comment by rvjansen** [comment]
I thought that the sixpack started out on a /380 enabled hercules, precisely because of the compiler

**Comment by BobBolch** [comment]
I believe the Sixpack and the Community Edition can all run in VM380 mode.
I don't know exactly how to do it yet.
Bob

**Comment by adesutherland** [comment]
I have merged develop into f0035. This includes some fixes to make the CMS to build. At least for the moment I have changed the gcc optimisation flag to -o1 (from -o2), which seems to have done the trick for the memory issue.

Rene/Rob/All - do you think we need to do a detour and work out how to rebuild gccmvs for 31 bit etc. Perhaps we need to do it for MVS as well? Meantime cREXX is building but ...
